src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/RestoreRunnerEx.cs

```diff
@@ -45,72 +44,85 @@ public static Task<IReadOnlyList<RestoreResultPair>> RunWithoutCommit(LibraryIde
                 IgnoreFailedSources = true,
             })
             {
-                var projectPath = Path.Combine(Path.GetTempPath(), TempProjectName);
+                // Create a unique temporary directory for the project
+                var projectDirectory = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString("N")));
```

Comment:
Should we use [NuGetScratch](https://github.com/NuGet/NuGet.Client/blob/cdf58815319ec0c9d1a29c3c790233826c1bab8a/src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs#L79) instead of just `Path.GetTempPath()`?

Reply:
Sure, I'll do that